staging/src/k8s.io/apiserver/pkg/server/config_selfclient_test.go

```diff
@@ -44,26 +44,43 @@ func TestLoopbackHostPort(t *testing.T) {
 		t.Fatalf("expected 443 as port, got %q", port)
 	}
 
-	host, port, err = LoopbackHostPort("[ff06:0:0:0:0:0:0:c3]:443")
-	if err != nil {
-		t.Fatalf("unexpected error: %v", err)
-	}
-	if host != "ff06:0:0:0:0:0:0:c3" {
-		t.Fatalf("expected ff06:0:0:0:0:0:0:c3 as host, got %q", host)
-	}
-	if port != "443" {
-		t.Fatalf("expected 443 as port, got %q", port)
-	}
+	if ipv6, err := isIPv6LoopbackSupported(); err == nil && ipv6 {
```

Comment:
please separate into another test and use `t.Skip`

Reply:
thank you @deads2k, fixed

Reply:
@deads2k i refreshed the pr to split the test but the test will still fail on conditions that don't support test run, although the improvement is to fail with very explicit error msg, 

@lavalamp raised a good question: we don't want to silently skipping tests in k/k CI, i agree with him considering we don't have a good way to monitor skipping tests.

user may still see test failure on those special host condition, but they should get very clear error message now, and can triage it accordingly.